dashboard/lib/services/teacher.rb

```diff
@@ -0,0 +1,15 @@
+module Services
+  module Teacher
```

Comment:
What's the rationale for choosing this particular module path, compared to say `Services::User` or `Services::User::Permissions`?

Reply:
The idea is to have separate modules for the teacher-specific and student-specific business logic.
I’ve wanted to implement this for a long time but never had the opportunity. Big thanks to @mgc1194 for making it happen!

Reply:
I see, and these modules could house lots of the teacher/student-specific logic that currently resides in the bloated User model? And for larger chunks of functionality, full-fledged service objects would be used but also live under `Services::Teacher` or `Services::Student`?

Reply:
That's a good question.
We already have a `Sevice::User` but to prevent it from becoming too big (like our user model) I wanted something more specific. While reading this [Toptal article](https://www.toptal.com/ruby-on-rails/rails-service-objects-tutorial) and [the honeybadger guide](https://www.honeybadger.io/blog/refactor-ruby-rails-service-object/), I understood that:

> Service objects are designed to execute one single action in your domain logic and do it well

My first instinct was to create a service called `TeacherVerifier`, however, Artem suggested creating separate namespaces for teachers and students, as the two account types have very different needs.

Reply:
It's true, but I would still call it `Services::Teacher::Verifier` to maintain the `Teacher` namespace

Reply:
Yeah, that sounds good. I can refactor the PR and add the syntactic sugar as in this example:
https://www.honeybadger.io/blog/refactor-ruby-rails-service-object/